### PR TITLE
feat(custom-convention): add custom convention to repository

### PR DESCRIPTION
**Pull request by maikmb**
## Proposed Changes

1. Include custom convention to read fields from database

## Example 

```
 const ItemRepository = new ItemRepository({
      entity: UserEntity,
      table,
      schema,
      ids: ['id'],
      knex: connection,
      convention: {
        toTableFieldName: field => toCamelCase(field)
      }
    })
```

### Author/Contributor
- [X] If documentation is needed for this change, has that been included in this pull request
- [ ] Remember to check if code coverage decrease, if so, please implement the necessary tests

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [X] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,14 +252,35 @@ const repo = new ItemRepository(injection)
 const ret = await repo.delete(entity)
 ```
 
-### Conventions - Defaul implementation
+## Conventions 
+
+### Defaul implementation
```
Comment (jhomarolo) [inline README.md:257]: please correct "Default"

File: README.md [hunk 2]
```diff
@@ -252,14 +252,35 @@ const repo = new ItemRepository(injection)
...
 
 #### Fields
 
 Code: Camel Case - ex: `productName`
```
Comment (jhomarolo) [inline README.md:261]: It's important to say why we choose this convention

File: src/convention.js [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
\ No newline at end of file
+  camelToSnake (string) {
+    return string.replace(/([A-Z])/g, '_$1').toLowerCase()
+  }
+
+  toTableFieldName (entityField) {
+    return this.camelToSnake(entityField)
+  }
+
+  isScalarType (type) {
```
Comment (jhomarolo) [inline src/convention.js:10]: does make sense this checker be inside suma?

File: src/dataMapper.js [hunk 4]
```diff
@@ -1,129 +1,129 @@
...
-
-    toEntity(payload) {
-        if (this._proxy === undefined) this._proxy = this.buildProxy()
-        this._proxy.load(payload)
-        return this.entity.fromJSON(this._proxy, { allowExtraKeys: true })
+  constructor (entity, entityIDs = [], foreignKeys = [], options = {}) {    
+    this.convention = Object.assign(new Convention(), options.convention)
+    this.entity = entity
+    const schema = entity.prototype.meta.schema
+    this.allFields = DataMapper.buildAllFields(
```
Comment (jhomarolo) [inline src/dataMapper.js:9]: Have you changed the lint? Why?

**Review by jhomarolo — CHANGES_REQUESTED** [review]
Hi @maikmb, could you please return back to the default lint/format of the project?

It's very hard do compare whats is new here.

**Comment by maikmb** [comment]
> Hi @maikmb, could you please return back to the default lint/format of the project?
> 
> It's very hard do compare whats is new here.

Hi @jhomarolo, to simplify the PR I have separated the refactoring part of the test structure #37

File: README.md (hunk 1 above)
Comment (maikmb) [reply README.md]: Fixed!

File: src/convention.js (hunk 3 above)
Comment (maikmb) [reply src/convention.js:10]: @jhomarolo this function was already used in the DataMapper Proxy.

https://github.com/maikmb/herbs2knex/blob/master/src/dataMapper.js#L93

File: src/dataMapper.js (hunk 4 above)
Comment (maikmb) [reply src/dataMapper.js]: Fixed!

File: src/convention.js (hunk 3 above)
Comment (jhomarolo) [reply src/convention.js:10]: I'll aprove this, but please open a issue in suma to transfer this checker there.

File: test/unit/queries/update.test.js [hunk 5]
```diff
@@ -148,8 +152,17 @@ describe("Update an Entity", () => {
     assert.deepStrictEqual(ret.id, 1)
     assert.deepStrictEqual(spy.where, 'id')
     assert.deepStrictEqual(spy.value, 1)
-    assert.deepStrictEqual(spy.fields, ['id', 'string_test', 'boolean_test', 'fk_field'])
-    assert.deepStrictEqual(spy.payload, { id: 1, string_test: 'test', boolean_test: true, fk_field: 42 })
+    assert.deepStrictEqual(spy.fields, [
```
Comment (jhomarolo) [inline test/unit/queries/update.test.js:155]: here still the old lint

**Review by jhomarolo — CHANGES_REQUESTED** [review]

**Pull request merged**